src/hooks/context/useTransactionsStatus.tsx

```diff
@@ -0,0 +1,112 @@
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import { TransactionStatusEnum, TransactionStatusType } from '../../types/transactionTypes';
+import { useUpdateMessageContext } from '../../contexts/MessageContext';
+
+import useChainGetters from '../chain/useChainGetters';
+import { useSubscriptionsContext } from '../../contexts/SubscriptionsContextProvider';
+import { getChainSubscriptionsKey } from '../../util/chainsUtils';
+import { handleTransactionUpdates } from '../../util/transactionUtils';
+
+import { useSourceTarget } from '../../contexts/SourceTargetContextProvider';
+import { useApiCallsContext } from '../../contexts/ApiCallsContextProvider';
+import useLaneId from '../chain/useLaneId';
+import { useEffect } from 'react';
+import { TransactionActionCreators } from '../../actions/transactionActions';
+import isEqual from 'lodash/isEqual';
+import { MessageActionsCreators } from '../../actions/messageActions';
+
+export default function useTransactionsStatus(
+  transactions: TransactionStatusType[],
+  evaluatingTransactions: boolean,
+  dispatchTransaction: Function
+) {
+  const { dispatchMessage } = useUpdateMessageContext();
+  const { getValuesByChain } = useChainGetters();
+
+  const subscriptions = useSubscriptionsContext();
+
+  const {
+    sourceChainDetails: { chain: currentSourceChain }
+  } = useSourceTarget();
+  const apiCalls = useApiCallsContext();
+  const laneId = useLaneId();
+
+  useEffect(() => {
+    const getTransactionStatus = async () => {
+      dispatchTransaction(TransactionActionCreators.setEvaluatingTransactionsStatus(true));
+      const updatedTransactions = await Promise.all(
+        transactions.map(async (transaction: TransactionStatusType) => {
+          if (transaction.status === TransactionStatusEnum.COMPLETED) {
+            return transaction;
+          }
+
+          const { sourceChain, targetChain } = transaction;
+          const { api: targetApi } = getValuesByChain(targetChain);
+          const { sourceRole, targetRole } = getChainSubscriptionsKey({
+            currentSourceChain,
+            sourceChain
+          });
+
+          const sourceSubscriptions = subscriptions[sourceRole];
+          const targetSubscriptions = subscriptions[targetRole];
+
+          const updatedTransaction = await handleTransactionUpdates({
+            transaction,
+            sourceSubscriptions,
+            targetSubscriptions,
+            apiCalls: {
+              targetApi,
+              ...apiCalls
+            },
+            laneId,
+            dispatchMessage
+          });
+
+          return updatedTransaction;
+        })
+      );
+
+      if (!isEqual(transactions, updatedTransactions)) {
+        dispatchTransaction(TransactionActionCreators.updateTransactionsStatus(updatedTransactions));
+      }
+
+      dispatchTransaction(TransactionActionCreators.setEvaluatingTransactionsStatus(false));
```

Comment:
This should be moved to `finally` block of the upper `try-catch` to make sure we set it to `false` in case of an error as well.

Reply:
I found out that moving the `setEvaluatingTransactionsStatus(false)` is incorrect because the  `getTransactionStatus` triggers an async function , but it is not waiting for the function to end , as a result `finally` block gets executed in the middle of the execution and mess up the process.
The solution to make sure that the evaluating will be managed correctly is by adding the `setEvaluatingTransactionsStatus(false)` call in the catch section.

By the way, the function `getTransactionStatus()` is called without the await because useEffect should not be async ( we discussed this before in previous PR)

Reply:
Then the whole `try-catch` is misleading and wrong then, because it does not handle all possible errors. We do need to handle all possible errors from async function, so please use `.catch` then, or wrap the whole content into async-aware `try-catch` (pretty sure we discussed error handling as well in previous PR).